plugin/src/main/java/org/codice/acdebugger/plugin/ACDebuggerPlugin.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.acdebugger.plugin;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codice.acdebugger.Main;
+
+@Mojo(name = "start", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
+public class ACDebuggerPlugin extends AbstractMojo {
+  @Parameter private boolean skip;
+
+  @Parameter private boolean remoteDebugging;
+
+  @Parameter(defaultValue = "localhost")
+  private String host;
+
+  @Parameter(defaultValue = "5005")
+  private String port;
+
+  @Parameter private boolean wait;
+
+  @Parameter(defaultValue = "1")
+  private int timeout;
+
+  @Parameter private boolean reconnect;
+
+  @Parameter private boolean continuous;
+
+  @Parameter private boolean admin;
+
+  @Parameter private boolean debug;
+
+  @Parameter private boolean service;
+
+  @Parameter private boolean fail;
+
+  @Parameter private boolean grant;
+
+  @Parameter(defaultValue = "true")
+  private boolean osgi;
+
+  @VisibleForTesting
+  ACDebuggerPlugin(Boolean skip, Boolean remoteDebugging) {
+    this.skip = skip;
+    this.remoteDebugging = remoteDebugging;
+  }
+
+  // Maven plugins need a default constructor if another constructor is defined
+  ACDebuggerPlugin() {}
+
+  @Override
+  public void execute() {
+    if (skip || remoteDebugging) {
+      getLog().info("skipping AC Debugger");
+      return;
+    }
+
+    List<String> arguments = buildArguments();
+    Runnable acdebugger = () -> Main.main(arguments.toArray(new String[0]));
```

Comment:
✏️ `arguments.toArray(new String[0])` is very inefficient as it creates an array which ends up being thrown out.
```suggestion
    Runnable acdebugger = () -> Main.main(arguments.toArray(new String[arguments.size()]));
```

Reply:
In modern JVMs (>=6) the toArray method with an empty size has been intrinsified.

From IntelliJ
> There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]). 

>In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.

>This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).`